scripts/src/cowidev/vax/incremental/taiwan.py

```diff
@@ -51,13 +50,15 @@ def _parse_pdf_link(self, soup) -> str:
     def _parse_table(self, url_pdf: str):
         dfs = self._parse_tables_all(url_pdf)
         df = dfs[0]
+
         # Sanity check
-        cols_wrong = df.columns.difference(self.columns_accepted)
-        if cols_wrong.any():
-            raise ValueError(f"There are some unknown columns: {cols_wrong}")
+        cols = df.columns
+        if not (len(cols) == 4 and cols[0] == "廠牌" and cols[1] == "劑次" and cols[2].endswith("接種人次") and cols[3].startswith("累計") and cols[3].endswith("接種人次")):
```

Comment:
Here I would suggest using [the walrus operator](https://docs.python.org/3/whatsnew/3.8.html#assignment-expressions) and regular expression instead:

```python
# Sanity check
if not (
    len(cols := df.columns) == 4
    and cols[0] == "廠牌"
    and cols[1] == "劑次"
    and cols[2].endswith("接種人次")
    and re.match(r"\d+\/\d+\-\d+\/\d+接種人次", cols[2])
    and re.match(r"累計至 \d+\/\d+接種人次", cols[3])
):
```

Reply:
@lucasrodes thanks for the advice.

We might have to change the patterns of `cols[2]` from:

```
   and re.match(r"\d+\/\d+\-\d+\/\d+接種人次", cols[2])
```

to:

```
   and re.match(r"(\d+\/\d+\-)?\d+\/\d+接種人次", cols[2])
```

As the first date stamp is only there on Mondays.

Here's a repo with past PDFs for references: https://github.com/profmike/tw-vaccine-stats/tree/main/data/tw.gov.cdc.vaccination_reports/daily